src.js

```diff
@@ -42,6 +42,23 @@ function sortArray(numbers) {
  * @param term
  */
 function isPalindrome(term) {
+	//Replace all spaces and punct. Changes to lowercase.
+	term = term.replace(/[.,\/#!$%\^&\*;:{}=\-_`~()]/g, "");
+	term = term.replace(/–/g, "")
+	term = term.replace(/ /g, "");
+	term = term.toLowerCase();
```

Comment:
When sanitizing input, it's better to "Keep only what you want", than to "Remove what you don't want". It is pretty easy to find characters that will bypass your sanitizer:

`A man, a plan, a canal – [Panama]` is the first one that came to me, but also any emoji... like mooncakes at noon:

```
it('mooncakes are not palindromes', () => {
  expect(isPalindrome('🥮noon🥮')).to.eql(true);   <--- fails
});
```

The "Keep only what you want" version would be:
```
    term.toLowerCase().replace(/[^a-z]/g, "");
```

Reply:
You're 100% correct. I didn't even think about doing it the other way. I was coding for the test cases rather than accounting for all cases.

Thanks!